src/expr.rs

```diff
@@ -1910,7 +1910,10 @@ fn choose_rhs<R: Rewrite>(
                     expr.rewrite(context, shape).map(|s| format!(" {}", s))
                 }
                 (None, None) => None,
-                (Some(orig_rhs), _) => Some(format!(" {}", orig_rhs)),
+                (Some(orig_rhs), _) => {
+                    let prefix = if orig_rhs.starts_with('\n') { "" } else { " " };
+                    Some(format!("{}{}", prefix, orig_rhs))
```

Comment:
This is to prevent the `error[internal]: left behind trailing whitespace` error from occurring in the scenario detailed [above](https://github.com/rust-lang/rustfmt/pull/3884/files#r338857310)